### PR TITLE
Changed all var to using const or let except for ColorThief

### PR DESCRIPTION
**Pull request by ibnujakaria**
I'm proposing this pull request because the use of `var` leads to strange problem as it shares the same variable even if we declare it multiple times.

The problem was: when I created multiple instance of `ColorThief` for different images, running it asynchronously in the same time, all of the instances always resulted the same color.

### CODE REVIEW

File: src/color-thief.js [hunk 1]
```diff
@@ -25,7 +25,7 @@
   with a set of helper functions.
 */
 
-var CanvasImage = function (image) {
+let CanvasImage = function (image) {
```
Comment (ItamarShDev) [inline src/color-thief.js]: better use `const`

File: src/color-thief.js [hunk 2]
```diff
@@ -59,8 +59,8 @@ var ColorThief = function () {};
  *
  * */
 ColorThief.prototype.getColor = function(sourceImage, quality) {
-    var palette       = this.getPalette(sourceImage, 5, quality);
-    var dominantColor = palette[0];
+    let palette       = this.getPalette(sourceImage, 5, quality);
```
Comment (ItamarShDev) [inline src/color-thief.js]: better use `const`

File: src/color-thief.js [hunk 3]
```diff
@@ -59,8 +59,8 @@ var ColorThief = function () {};
...
+    let dominantColor = palette[0];
```
Comment (ItamarShDev) [inline src/color-thief.js]: maybe just return `palette[0]`

File: src/color-thief.js [hunk 4]
```diff
@@ -92,14 +92,14 @@ ColorThief.prototype.getPalette = function(sourceImage, colorCount, quality) {
     }
 
     // Create custom CanvasImage object
-    var image      = new CanvasImage(sourceImage);
-    var imageData  = image.getImageData();
-    var pixels     = imageData.data;
-    var pixelCount = image.getPixelCount();
+    let image      = new CanvasImage(sourceImage);
```
Comment (ItamarShDev) [inline src/color-thief.js]: all 4 should be const IMHO

File: src/color-thief.js [hunk 5]
```diff
@@ -115,18 +115,18 @@ ColorThief.prototype.getPalette = function(sourceImage, colorCount, quality) {
 
     // Send array to quantize function which clusters values
     // using median cut algorithm
-    var cmap    = MMCQ.quantize(pixelArray, colorCount);
-    var palette = cmap? cmap.palette() : null;
+    let cmap    = MMCQ.quantize(pixelArray, colorCount);
```
Comment (ItamarShDev) [inline src/color-thief.js]: should be const

File: src/color-thief.js [hunk 6]
```diff
@@ -115,18 +115,18 @@ ColorThief.prototype.getPalette = function(sourceImage, colorCount, quality) {
...
+    let palette = cmap? cmap.palette() : null;
```
Comment (ItamarShDev) [inline src/color-thief.js]:
```suggestion
    return cmap? cmap.palette() : null;
```

File: src/color-thief.js [hunk 7]
```diff
@@ -115,18 +115,18 @@ ColorThief.prototype.getPalette = function(sourceImage, colorCount, quality) {
...
 
     return palette;
 };
 
 ColorThief.prototype.getColorFromUrl = function(imageUrl, callback, quality) {
     let sourceImage = document.createElement("img");
-    var thief = this;
+    let thief = this;
```
Comment (ItamarShDev) [inline src/color-thief.js]: const

File: src/color-thief.js [hunk 8]
```diff
@@ -115,18 +115,18 @@ ColorThief.prototype.getPalette = function(sourceImage, colorCount, quality) {
...
     sourceImage.addEventListener('load' , function(){
-        var palette = thief.getPalette(sourceImage, 5, quality);
-        var dominantColor = palette[0];
+        let palette = thief.getPalette(sourceImage, 5, quality);
```
Comment (ItamarShDev) [inline src/color-thief.js]: const

File: src/color-thief.js [hunk 9]
```diff
@@ -115,18 +115,18 @@ ColorThief.prototype.getPalette = function(sourceImage, colorCount, quality) {
...
+        let dominantColor = palette[0];
```
Comment (ItamarShDev) [inline src/color-thief.js]:
```suggestion
        callback(palette[0], imageUrl);
```

File: src/color-thief.js [hunk 10]
```diff
@@ -140,9 +140,9 @@ ColorThief.prototype.getImageData = function(imageUrl, callback) {
     xhr.onload = function() {
         if (this.status == 200) {
             let uInt8Array = new Uint8Array(this.response)
-            i = uInt8Array.length
+            let i = uInt8Array.length
```
Comment (ItamarShDev) [inline src/color-thief.js]: seems redundant.

**Comment by ItamarShDev** [comment]
Basically, seems like most places should be const and not let.

**Comment by ibnujakaria** [comment]
I will try to make changes as your suggestions

**Comment by ItamarShDev** [comment]
looks good, don't forget to fix conflicts

**Comment by ibnujakaria** [comment]
Thanks.

Looks like the conflict has been solved.

**Comment by lokesh** [comment]
Sorry for all the delays and thank you for your work. The code has changed substantially in the past month as I've added new dist targets, node support, and move the clustering/quantize algorithm to a separate repo.

I reviewed your changes to the core color thief code in this PR and moved them into a new PR which I have merged: https://github.com/lokesh/color-thief/pull/168

The quantize code is now in the following repo: https://github.com/lokesh/quantize. I'm going to work on the algorithm code in the next few weeks after finishing the review of all existing Color Thief issues and PRs. At that time I'll come back and reference this PR.

Again, thanks for your effort and patience.

**Pull request closed without merging**